### PR TITLE
[RLlib] Make the "tiny CNN" example RLModule run with APPO (by implementing `TargetNetAPI`)

### PR DESCRIPTION
**Pull request by sven1977**
Make the "tiny CNN" example RLModule run with APPO (by implementing `TargetNetAPI`)

## Checks

- [x] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Review by simonsays1980 — APPROVED** [review]
LGTM. I have a question to the key used for the action logits.

File: rllib/examples/rl_modules/classes/tiny_atari_cnn_rlm.py [hunk 1]
```diff
@@ -141,6 +145,29 @@ def _forward_train(self, batch, **kwargs):
...
+            (self._logits, self._target_logits),
+        ]
+
+    @override(TargetNetworkAPI)
+    def forward_target(self, batch, **kw):
+        obs = batch[Columns.OBS].permute(0, 3, 1, 2)
+        embeddings = self._target_base_cnn_stack(obs)
+        logits = self._target_logits(embeddings)
+        return {
+            OLD_ACTION_DIST_LOGITS_KEY: torch.squeeze(logits, dim=[-1, -2])
```
[the remark this answers is not in the document]
Comment (sven1977) [reply rllib/examples/rl_modules/classes/tiny_atari_cnn_rlm.py]: Great point! This should probably not be algo-specific, either.
We should call it "TARGET_ACTION_DIST_INPUTS" to make it generic, and move it somewhere else (in the rl_module/apis folder).

Will change ...
